src/Functions/IFunction.h

```diff
@@ -258,6 +258,9 @@ class IFunctionBase : public IResolvedFunction
         /// Example: toTypeName(expr), even if expr contains functions that are not suitable for
         /// lazy execution (because of their simplicity), we shouldn't execute them at all.
         bool force_enable_lazy_execution;
+        /// Currently only useful for dictGetOrDefault, the 3rd argument of it should always be
+        /// calculated, so set it to false in dictGetOrDefault, while true in other cases.
+        bool enable_lazy_execution_for_third_argument = true;
```

Comment:
As now we have `enable_lazy_execution_for_first_argument` and `enable_lazy_execution_for_third_argument` maybe we can have single field with set of arguments that should not be executed lazily. And I guess for `dictGetOrDefault` we should not execute all first, second and third arguments.

Reply:
I get confused about "single field with set of arguments that should not be executed lazily", would you mind give an example?

Reply:
Like, one field `std::unordered_set<size_t> arguments_with_disabled_lazy_execution` instead of two fields `enable_lazy_execution_for_first_argument /enable_lazy_execution_for_third_argument `